File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        let seen = new Set(nums)
        let maxLen = 0;

        for(let num of nums){
            if(!seen.has(num-1)){
                let curr = num
                let len = 1
                while(seen.has(curr+1)){
                    curr++
                    len++
                }
                maxLen = Math.max(len, maxLen)
            }
        }
        return maxLen
    }
}

File: Data Structures & Algorithms/sliding-window-maximum/submission-3.cpp
class Solution {
public:
    vector<int> maxSlidingWindow(vector<int>& nums, int k) {
            int l=0;
            int r=0;
            vector<int> res;
            int max = INT_MIN;
            while(r<nums.size()){
                if(nums[r]>max){
                    max = nums[r];
                }
                if((r-l+1)==k){
                    res.push_back(max);
                    r=l;
                    l++;
                    max=INT_MIN;
                    
                   
                }   
               
                r++;
            }

        return res;
    }
};
